requests: Allow files={"example": ("filename", bytes)}

According to https://github.com/psf/requests/blob/2d5517682b3b38547634d153cea43d48fbc8cdb5/requests/models.py#L170 It should be possible to provide `files={"fieldname": ("filename", [bytes-object])}`, but since the last update of mypy this fails with
```
packet_analyser/daemon/uploader.py:78: error: Argument "files" to "post" has incompatible type "Dict[str, Tuple[str, bytes]]"; expected "Optional[Union[MutableMapping[str, IO[Any]], MutableMapping[str, Tuple[Optional[str], IO[Any]]], MutableMapping[str, Tuple[Optional[str], IO[Any], str]], MutableMapping[str, Tuple[Optional[str], IO[Any], str, MutableMapping[str, str]]]]]"  [arg-type]
```

The issue seems to be in https://github.com/python/typeshed/blob/63fb9af7438613dd2bfebe07911e49022ce78f57/stubs/requests/requests/sessions.pyi#L51-L56

Can this be fixed?

I see a similar issue for the example from https://docs.python-requests.org/en/latest/user/quickstart/#post-a-multipart-encoded-file:

```python
import requests

url = "https://httpbin.org/post"
files = {"file": open("report.xls", "rb")}

r = requests.post(url, files=files)
```

> bug.py:6: error: Argument "files" to "post" has incompatible type "Dict[str, BufferedReader]"; expected "Optional[Union[MutableMapping[str, IO[Any]], MutableMapping[str, Tuple[str, IO[Any]]], MutableMapping[str, Tuple[str, IO[Any], str]], MutableMapping[str, Tuple[str, IO[Any], str, MutableMapping[str, str]]]]]"  [arg-type]

This issue reproduces with `mypy==0.942` and `types-requests==2.27.21`, but not with `types-requests==2.27.20`.

I'm also seeing this issue when utilizing `StringIO` in a test:

```
    response = client.put(
        f"{BASE_PATH}/name/file-import",
        files={"file": StringIO("mockfile")},
    )

error: Argument "files" to "post" has incompatible type "Dict[str, Tuple[str, bytes]]"; expected "Optional[Union[MutableMapping[str, IO[Any]], MutableMapping[str, Tuple[Optional[str], IO[Any]]], MutableMapping[str, Tuple[Optional[str], IO[Any], str]], MutableMapping[str, Tuple[Optional[str], IO[Any], str, MutableMapping[str, str]]]]]"  [arg-type]
```

--- a/stubs/requests/requests/sessions.pyi
+++ b/stubs/requests/requests/sessions.pyi
@@ -49,10 +49,10 @@ _Data: TypeAlias = str | bytes | Mapping[str, Any] | Iterable[tuple[str, str | N
 _Auth: TypeAlias = Union[tuple[str, str], _auth.AuthBase, Callable[[PreparedRequest], PreparedRequest]]
 _Cert: TypeAlias = Union[str, tuple[str, str]]
 _Files: TypeAlias = (
-    MutableMapping[str, IO[Any]]
-    | MutableMapping[str, tuple[str | None, IO[Any]]]
-    | MutableMapping[str, tuple[str | None, IO[Any], str]]
-    | MutableMapping[str, tuple[str | None, IO[Any], str, _TextMapping]]
+    MutableMapping[str, IO[Any] | str | bytes]
+    | MutableMapping[str, tuple[str | None, IO[Any] | str | bytes]]
+    | MutableMapping[str, tuple[str | None, IO[Any] | str | bytes, str]]
+    | MutableMapping[str, tuple[str | None, IO[Any] | str | bytes, str, _TextMapping]]
 )
 _Hook: TypeAlias = Callable[[Response], Any]
 _Hooks: TypeAlias = MutableMapping[str, _Hook | list[_Hook]]

--- a/stubs/requests/requests/sessions.pyi
+++ b/stubs/requests/requests/sessions.pyi
@@ -1,4 +1,4 @@
-from _typeshed import Self, SupportsItems
+from _typeshed import Self, SupportsItems, SupportsRead
 from collections.abc import Callable, Iterable, Mapping, MutableMapping
 from typing import IO, Any, Union
 from typing_extensions import TypeAlias
@@ -49,10 +49,10 @@ _Data: TypeAlias = str | bytes | Mapping[str, Any] | Iterable[tuple[str, str | N
 _Auth: TypeAlias = Union[tuple[str, str], _auth.AuthBase, Callable[[PreparedRequest], PreparedRequest]]
 _Cert: TypeAlias = Union[str, tuple[str, str]]
 _Files: TypeAlias = (
-    MutableMapping[str, IO[Any] | str | bytes]
-    | MutableMapping[str, tuple[str | None, IO[Any] | str | bytes]]
-    | MutableMapping[str, tuple[str | None, IO[Any] | str | bytes, str]]
-    | MutableMapping[str, tuple[str | None, IO[Any] | str | bytes, str, _TextMapping]]
+    MutableMapping[str, SupportsRead[str | bytes] | str | bytes]
+    | MutableMapping[str, tuple[str | None, SupportsRead[str | bytes] | str | bytes]]
+    | MutableMapping[str, tuple[str | None, SupportsRead[str | bytes] | str | bytes, str]]
+    | MutableMapping[str, tuple[str | None, SupportsRead[str | bytes] | str | bytes, str, _TextMapping]]
 )
 _Hook: TypeAlias = Callable[[Response], Any]
 _Hooks: TypeAlias = MutableMapping[str, _Hook | list[_Hook]]

